src/nvim/event/libuv_process.c

```diff
@@ -61,8 +61,9 @@ int libuv_process_spawn(LibuvProcess *uvproc)
   if (!proc->out.closed) {
     uvproc->uvstdio[1].flags = UV_CREATE_PIPE | UV_WRITABLE_PIPE;
 #ifdef WIN32
-    // pipe must be readable for IOCP to work.
-    uvproc->uvstdio[1].flags |= UV_READABLE_PIPE | UV_OVERLAPPED_PIPE;
+    // pipe must be readable for IOCP to work on Windows.
+    uvproc->uvstdio[1].flags |= proc->overlapped ?
+      (UV_READABLE_PIPE | UV_OVERLAPPED_PIPE) : 0;
```

Comment:
Should `UV_READABLE_PIPE` be set even if `proc->overlapped` is false?

Reply:
Sorry, I missed this. This is only required for `pynvim` for the reasons stated in https://github.com/neovim/pynvim/pull/307#issuecomment-375265729. Therefore, if `proc->overlapped` is `false`, `UV_READABLE_PIPE` is not required.

Should this be a separate option?

Reply:
> This is only required for `pynvim` for the reasons stated in [neovim/pynvim#307 (comment)](https://github.com/neovim/pynvim/pull/307#issuecomment-375265729). Therefore, if `proc->overlapped` is `false`, `UV_READABLE_PIPE` is not required.

Thanks for the explanation.

> Should this be a separate option?

Not unless we have a use-case for it.